lib/semian/sysv_shared_memory.rb

```diff
@@ -0,0 +1,75 @@
+module Semian
+  module SysVSharedMemory #:nodoc:
+    @type_size = {}
+    def self.sizeof(type)
+      size = (@type_size[type.to_sym] ||= (respond_to?(:_sizeof) ? _sizeof(type.to_sym) : 0))
+      raise TypeError.new("#{type} is not a valid C type") if size <= 0
+      size
+    end
+
+    def self.included(base)
+      def base.do_with_sync(*names)
```

Comment:
While looking over the C code base, one of the scarier things was that because I provided the lock() and unlock(), it was very difficult to track where I had to re-unlock, since errors could occur in the middle of a function, and then you'd have to insert an unlock() there. I then opted to write `ensures` in, but doing so manually would add twice the number of functions and make things very manual. 

Here's the c-ensure:
```C
return rb_ensure(semian_shm_object_synchronize_with_block, self, semian_shm_object_synchronize_restore_lock_status, (VALUE)&status);
```
Essentially each function needed a wrapper, and probably also some custom struct to overcome the one-argument limit of the function. This naturally lead to me doing it in Ruby instead since it doesn't have these limitations, which is why `synchronize` does what it does. IMO it's not a big deal since it only does it once per class, reduces code and complexity by a bit.

With regards to the noop `synchronize`, do you mean something more along the lines of (1) , with SharedMemoryObject -> Simple::Integer -> SysV::Integer ? Otherwise I don't understand what you mean by noop `synchronize`. Unless you mean just having one there to keep the interfaces the same?